src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -0,0 +1,89 @@
+package is.hail.variant
+
+import is.hail.expr.{JSONAnnotationImpex, TInterval}
+import is.hail.utils._
+import org.json4s._
+import org.json4s.jackson.JsonMethods
+
+import scala.reflect.ClassTag
+
+case class GenomeReference(name: String, contigs: Array[Contig], xContigs: Set[String],
+  yContigs: Set[String], mtContigs: Set[String], par: Array[Interval[Locus]]) extends Serializable {
+
+  def inX(contig: String): Boolean = xContigs.contains(contig)
+
+  def inY(contig: String): Boolean = yContigs.contains(contig)
+
+  def isMitochondrial(contig: String): Boolean = mtContigs.contains(contig)
+
+  def inXPar(locus: Locus): Boolean = inX(locus.contig) && par.exists(_.contains(locus))
+
+  def inYPar(locus: Locus): Boolean = inY(locus.contig) && par.exists(_.contains(locus))
+}
+
+object GenomeReference {
+  def GRCh37 = fromResource("reference/human_g1k_v37.json")
+
+  def fromResource(file: String): GenomeReference = {
+    val resourceStream = Thread.currentThread().getContextClassLoader.getResourceAsStream(file)
+
+    try {
+      if (resourceStream == null) {
+        throw new RuntimeException(s"Could not read genome reference file `$file'.")
+      }
+
+      val json = JsonMethods.parse(resourceStream)
+
+      val fields = json.asInstanceOf[JObject].obj.toMap
```

Comment:
I always thought the metadata code was a mess.  I think we can do better.  I'd use `JValue.extract` to load the data into a case class.  See "Extracting Values" here: http://json4s.org/.